inttest/capi-docker-machinedeployment/capi_docker_test.go

```diff
@@ -150,8 +169,88 @@ func (s *CAPIDockerSuite) TestCAPIDocker() {
 	}))
 }
 
-func (s *CAPIDockerSuite) applyClusterObjects() {
+func (s *CAPIDockerSuite) applyClusterObjects(yamlPath string) {
 	// Exec via kubectl
-	out, err := exec.Command("kubectl", "apply", "-f", "../../config/samples/capi/docker/cluster-with-machinedeployment.yaml").CombinedOutput()
+	out, err := exec.Command("kubectl", "apply", "-f", yamlPath).CombinedOutput()
 	s.Require().NoError(err, "failed to apply cluster objects: %s", string(out))
 }
+
+// getK0smotronControlPlaneStatus retrieves the K0smotronControlPlane status
+func (s *CAPIDockerSuite) getK0smotronControlPlaneStatus(name, namespace string) (*cpv1beta1.K0smotronControlPlaneStatus, error) {
+	kcp := &cpv1beta1.K0smotronControlPlane{}
+	err := s.client.RESTClient().
+		Get().
+		AbsPath("/apis/controlplane.cluster.x-k8s.io/v1beta1").
+		Resource("k0smotroncontrolplanes").
+		Namespace(namespace).
+		Name(name).
+		Do(s.ctx).Into(kcp)
+	if err != nil {
+		return nil, err
+	}
+	return &kcp.Status, nil
+}
+
+// verifyK0smotronControlPlaneVersionFormat verifies that the version format is consistent
+func (s *CAPIDockerSuite) verifyK0smotronControlPlaneVersionFormat(clusterName, expectedSpecVersion string, expectK0sSuffix bool) error {
```

Comment:
I think it's too complex func for the task. Could we just keep it simple?

Reply:
Thx. I did it.